### ISSUE TITLE
Manager: writes garbage to cc_config.xml

### ISSUE THREAD

**Issue by RichardHaselgrove** [BOINC/boinc#3247]
**Describe the bug**
When the 'Event log options...' dialog is used to modify the active set of Event Log flags, garbage is written to cc_config.xml.

**Steps To Reproduce**
1. Open Event Log Options dialog, change a setting, apply or save your changes
2. Inspect the contents of cc_config.xml visually
3. (optional) Shut down and restart the BOINC client...

**Expected behavior**
Existing option settings are left unchanged'

**Screenshots**
Original entries in cc_config.xml:
```
    <options>
        <abort_jobs_on_exit>0</abort_jobs_on_exit>
        <allow_multiple_clients>1</allow_multiple_clients>
        <allow_remote_gui_rpc>0</allow_remote_gui_rpc>
        <disallow_attach>0</disallow_attach>
```

New entries in cc_config.xml:
```
    <options>
        <abort_jobs_on_exit>1069290768</abort_jobs_on_exit>
        <allow_gui_rpc_get>0</allow_gui_rpc_get>
        <disallow_attach>0</disallow_attach>
```

**System Information (please complete the following information):**
 - OS: Windows, Linux (Mac untested)
 - BOINC Version: 7.16.1 (new in this version)

**Additional context**
Possibly came about as a side effect of #3214 (added <allow_gui_rpc_get>), but I haven't tracked it down yet - I'll look over the weekend.

With a value >0 in abort_jobs_on_exit - guess what happens on exit! That's the blocker.

Problem was noticed as a result of using the **release** version of 7.16.1 from @LocutusOfBorg PPA, verified with self-build version of boincmgr for Windows (no test version available yet).

**Comment by RichardHaselgrove** [comment]
Further testing reinforces my belief that this is a Manager problem. Sending the 'update log flags' RPC remotely from an older version Manager doesn't corrupt the cc_config.xml file. But #3214 didn't modify the manager, and the changes to  lib/cc_config.cpp look benign. Still investigating.

**Comment by RichardHaselgrove** [comment]
I've been trying to trace back this problem, to find out for certain when and where it started. Bear in mind that (a) Appveyor builds when the PR is created, not when it is merged, and (b) Appveyor wasn't working for a long time. No artifacts were produced for PR 3214.

So I've tested the artifacts for PR 3230 (built July 20, before 3214 was merged) and PR 3233 (built July 26).

The client from PR 3233 behaves normally (no corruption, with the older manager), but the Manager from PR 3233 (with the matching client) introduces the corruption documented here.
